apps/web/pages/getting-started.tsx

```diff
@@ -662,59 +704,18 @@ export async function getServerSideProps(context: NextPageContext) {
     };
   }
 
-  const credentials = await prisma.credential.findMany({
-    where: {
-      userId: user.id,
-    },
-    select: {
-      id: true,
-      type: true,
-      key: true,
-      userId: true,
-      appId: true,
-    },
-  });
-
-  const integrations = getApps(credentials)
+  const integrations = getApps(user.credentials)
     .filter((item) => item.type.endsWith("_calendar"))
     .map((item) => omit(item, "key"));
-
-  // get user's credentials + their connected integrations
-  const calendarCredentials = getCalendarCredentials(credentials, user.id);
-  // get all the connected integrations' calendars (from third party)
-  const connectedCalendars = await getConnectedCalendars(calendarCredentials, user.selectedCalendars);
-
-  const eventTypes = await prisma.eventType.findMany({
-    where: {
-      userId: user.id,
-    },
-    select: {
-      id: true,
-      title: true,
-      slug: true,
-      description: true,
-      length: true,
-      hidden: true,
-    },
-  });
-
-  const schedules = await prisma.schedule.findMany({
-    where: {
-      userId: user.id,
-    },
-    select: {
-      id: true,
-    },
-  });
+  const { schedules } = user;
+  const hasConfigureCalendar = integrations.some((integration) => integration.credential !== null);
+  const hasSchedules = schedules && schedules.length > 0;
 
   return {
     props: {
       session,
       user,
-      integrations,
-      connectedCalendars,
-      eventTypes,
-      schedules,
+      initialStep: hasSchedules ? (hasConfigureCalendar ? 2 : 3) : 0,
```

Comment:
Only passing necessary data to the frontend. Reducing HTML size and possible data leakage